glom/mutable.py

```diff
@@ -0,0 +1,63 @@
+'''
+this module contains Specs that perform mutations
+'''
+from .core import _TType, _T_PATHS, _t_child, _t_eval, Path, T, S, Spec, BaseSpec, glom
+
+from . import core
+
+if 'basestring' in core.__dict__:
+    from .core import basestring
+
+# TODO: how to get this into default spec registry?
+# options:
+#   1- have a different registry here
+#   2- just eat the circularity
+#   3- move Assign back to core.py (but that is getting large)
+#   4- add some default behavior -- "if it is an instance of BaseSpec, call handle"
+#   5- class decorator that does a bit of checking as well as the registration
+class Assign(BaseSpec):
+    def __init__(self, path, val):
+        if isinstance(path, basestring):
+            path = Path(*path.split('.')).path_t
+        elif type(path) is Path:
+            path = path.path_t
+        elif not isinstance(path, _TType):
+            raise TypeError('path argument must be a .-delimited string, Path, T, or S')
+
+        segs = _T_PATHS[path]
+        if len(segs) < 3:
+            raise ValueError('path must have at least one element')
+
+        cur = segs[0]
+        assert cur in (T, S)
+        for i in range(1, len(segs) - 2, 2):
+            cur = _t_child(cur, segs[i], segs[i + 1])
+        self.t = cur
+        self.op, self.arg = segs[-2:]
+        if self.op not in '[.P':  # maybe if we add null-coalescing this should do something?
```

Comment:
the idea here is not to say "is this a supported operator for T?" -- the operator just came out of a T instance so we can assume so

the idea is, "is this an operator that will definitely give a result?"  e.g. if we have "?" that does null-coalescing, then it isn't safe for the first argument because we would end up trying to set an attribute on None

Reply:
e.g. `Assign('a?b.c', 1)` would be a problem, because `a?b.c` would result in `None` if 'b' doesn't exist

Reply:
I think the semantics of `?` for set would be to skip setting, because we generally can't create intermediaries. Maybe we have a default factory hook that takes a path or something?